src/api_manager/context/global_context.cc

```diff
@@ -148,6 +148,21 @@ GlobalContext::CreateCloudTraceAggregator() {
       minimum_qps, env_.get()));
 }
 
+void GlobalContext::AddInstanceIdentityToken(
+    const std::string& audience,
+    std::unique_ptr<auth::ServiceAccountToken> token) {
+  instance_identity_token_map_.emplace(audience, std::move(token));
+}
+
+auth::ServiceAccountToken* GlobalContext::GetInstanceIdentityToken(
+    const std::string& audience) {
```

Comment:
maybe combine two functions into one Get function.  If audience is not found, just create a new entry.

Reply:
Done.